atc/engine/check_delegate_test.go

```diff
@@ -440,6 +440,37 @@ var _ = Describe("CheckDelegate", func() {
 					Expect(run).To(BeFalse())
 				})
 			})
+
+			Context("when the checking interval has elapsed since the last check end time", func() {
+				BeforeEach(func() {
+					fakeResourceConfigScope.LastCheckReturns(db.LastCheck{
+						StartTime: now.Add(-6 * time.Minute),
+						EndTime:   now.Add(-5 * time.Minute),
+						Succeeded: true,
+					}, nil)
+					fakeBuild.StartTimeReturns(now)
+				})
+
+				It("returns true", func() {
+					Expect(run).To(BeTrue())
+				})
+			})
+
+			Context("when the checking interval has not elapsed since the last check end time", func() {
+				BeforeEach(func() {
+					plan.Check.Interval.Interval = time.Hour
+					fakeResourceConfigScope.LastCheckReturns(db.LastCheck{
+						StartTime: now,
+						EndTime:   now,
+						Succeeded: true,
```

Comment:
Oh I actually made it so that the last check end time is before the build start time so that it won't return false because of the first part of the `OR`. But if you think adding this is better than I will add it!

Reply:
I see. Guess I just want to make sure which taks priority between last check status and elapsed time in the `if` clause. In above test i see it test with last check status false without considerate elapse time interval.

https://github.com/concourse/concourse/blob/c6e9838023a40599d41d7d1a7aabf057ead03d71/atc/engine/check_delegate_test.go#L400-L412